xgcm/grid_ufunc.py

```diff
@@ -599,25 +598,43 @@ def _get_chunk_pattern_for_merging_boundary(
     da: xr.DataArray,
     original_chunks: Mapping[str, Tuple[int, ...]],
     boundary_width_real_axes: Mapping[str, Tuple[int, int]],
-):
+) -> Mapping[str, Tuple[int, ...]]:
     """Calculates the pattern of chunking needed to merge back in small chunks left on boundaries after padding"""
 
-    # Easier to work with width of boundaries in terms of str axis names rather than int axis numbers
+    # Easier to work with width of boundaries in terms of str dimension names rather than int axis numbers
     boundary_width_dims = {
         _get_dim(grid, da, ax): width for ax, width in boundary_width_real_axes.items()
     }
 
     new_chunks = {}
     for dim, width in boundary_width_dims.items():
         lower_boundary_width, upper_boundary_width = boundary_width_dims[dim]
-        first_chunk_width, *other_chunks_widths, last_chunk_width = original_chunks[dim]
-        new_chunks_along_dim = tuple(
-            [
-                first_chunk_width + lower_boundary_width,
-                *other_chunks_widths,
-                last_chunk_width + upper_boundary_width,
+
+        if len(original_chunks[dim]) == 1:
+            # unpadded array had only one chunk, but padding has meant new array is extended
+            original_array_length = original_chunks[dim][0]
+            new_chunks_along_dim = (
+                lower_boundary_width + original_array_length + upper_boundary_width
+            )
+        elif len(original_chunks[dim]) == 2:
+            first_chunk_width, last_chunk_width = original_chunks[dim]
+            new_chunks_along_dim = tuple(
+                [
+                    first_chunk_width + lower_boundary_width,
+                    last_chunk_width + upper_boundary_width,
+                ]
+            )
```

Comment:
I might be able to get rid of this by making `other_chunks_widths` an empty list in this case.